dataprofiler/tests/profilers/test_unstructured_text_profile.py

```diff
@@ -164,4 +165,118 @@ def test_case_sensitivity(self):
             expected_word_count = {'hello': 1, 'name': 1, 'grant': 2, 'bob': 1,
                                    'friends': 1}
             self.assertDictEqual(expected_word_count, profile['word_count'])
+
+    def test_options_text_profile(self):
+        # option check for the invalid values
+        options_case_sensitive = [2, 'string']
+        for option_case_sensitive in options_case_sensitive:
+            options = TextProfilerOptions()
+            options.is_case_sensitive = option_case_sensitive
+            with self.assertRaisesRegex(
+                    ValueError,
+                    "TextProfilerOptions.is_case_sensitive must be a Boolean."):
+                TextProfiler("Name", options=options)
+
+        options_stop_words = [2, 'a', [1, 2]]
+        for option_stop_words in options_stop_words:
+            options = TextProfilerOptions()
+            options.stop_words = option_stop_words
+            with self.assertRaisesRegex(
+                    ValueError,
+                    "TextProfilerOptions.stop_words must be None "
+                              "or list of strings."):
+                TextProfiler("Name", options=options)
+
+        options_words = [2, True]
+        for option_words in options_words:
+            options = TextProfilerOptions()
+            options.words = option_words
+            with self.assertRaisesRegex(
+                    ValueError,
+                    "TextProfilerOptions.words must be a BooleanOption "
+                          "object."):
+                TextProfiler("Name", options=options)
+
+        options_vocab = [2, True]
+        for option_vocab in options_vocab:
+            options = TextProfilerOptions()
+            options.vocab = option_vocab
+            with self.assertRaisesRegex(
+                    ValueError,
+                    "TextProfilerOptions.vocab must be a BooleanOption "
+                    "object."):
+                TextProfiler("Name", options=options)
+
+        # default options
+        options = TextProfilerOptions()
+
+        text_profile = TextProfiler("Name", options=options)
+        sample = pd.Series(["This is, a Test sentence.!!!"])
+        text_profile.update(sample)
+        profile = text_profile.profile
+
+        expected_word_count = {'sentence': 1, 'Test': 1}
+        expected_vocab = [' ', ',', '.', '!', 'T', 'h', 'i', 's', 'a',
+                          'e', 't', 'n', 'c']
+        self.assertDictEqual(expected_word_count, profile['word_count'])
+        self.assertCountEqual(expected_vocab, profile['vocab'])
+
+        # is_case_sensitive options
```

Comment:
Can you add a test where the data is case sensitive, and there are 2 words that are the same, but one is capitalized, and ensure that it still treats them as 2 separate words?

Reply:
good point! I've added it